Can't run on NixOS unstable (No Screen Found and nix errors)

Can't run any program with `primusrun` straight from the repo (as script). 
It shows errors on build and fails with `no screens found`.
Running Nitro 5 with Intel i5-8300H and GTX1050.

Console: https://hastebin.com/momuqifaci.sql
Xorg: https://hastebin.com/alatekofaf.sql

Nix-info output:
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.14, NixOS, 19.03pre166331.f9bd1533773 (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-18.09.1922.97e0d53d669, nixpkgs-19.03pre166331.f9bd1533773"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```
Building linux driver against kernel: /nix/store/3czhbb7670xy1zalrg2nw98vh95si35j-linux-4.14.93-dev
make[1]: *** /nix/store/3czhbb7670xy1zalrg2nw98vh95si35j-linux-4.14.93-dev/lib/modules/4.19.14/source: No such file or directory.  Stop.
make: *** [Makefile:79: modules] Error 2
```
A reboot is required after system update or kernel version change for this script to work.
unstable recently switched from 4.19 back to 4.14

After reboot console log still contain the following error?
```
Building linux driver against kernel: /nix/store/3czhbb7670xy1zalrg2nw98vh95si35j-linux-4.14.93-dev
make[1]: *** /nix/store/3czhbb7670xy1zalrg2nw98vh95si35j-linux-4.14.93-dev/lib/modules/4.19.14/source: No such file or directory.  Stop.
make: *** [Makefile:79: modules] Error 2
```
1. If no, upload new logs
2. If yes, show the output of following commands
`uname -r`
`nix-build --no-out-link -E 'with import <nixpkgs/nixos> {}; config.boot.kernelPackages.kernel.dev`
E.g.
`4.19.20`
`/nix/store/hzjdmj01sng0gla0habqjmqq8ckmjyxp-linux-4.19.20-dev`

--- a/primerun/primerun.sh
+++ b/primerun/primerun.sh
@@ -67,6 +67,10 @@ if [ "$busid" == "" ]; then
   echo -e "$red No Optimus card found"
   exit 1
 fi
+if [ ! -d "$(nix-build --no-out-link -E 'with import <nixpkgs/nixos> {}; config.boot.kernelPackages.kernel.dev')/lib/modules/$kernel" ]; then
+  echo -e "$red There is a mismatch between config.boot.kernelPackages.kernel: $(nix-instantiate --eval --strict -E '(import <nixpkgs/nixos> {}).config.boot.kernelPackages.kernel.dev.version') and currently running kernel: $kernel. Reboot is required."
+  exit 1
+fi
 # module nvidia_drm is in use
 if [ "$(sudo fgconsole)" == "$console" ]; then
   echo -e "$red You need to run $0 from another console"

